sources/api/apiserver/src/server/controller.rs

```diff
@@ -449,13 +589,135 @@ pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+        .get_metadata_prefix("settings.", committed, &None as &Option<&str>)
+        .context(error::DataStoreSnafu {
+            op: "get_metadata_prefix",
+        })?;
+
+    // Vector(metadata_key, key, value)
+    let mut metadata_to_commit: Vec<(Key, Key, String)> = Vec::new();
+
+    // Parse and validate all the metadata enteries from pending transaction
+    for (key, value) in transaction_metadata.iter_mut() {
+        for (metadata_key, metadata_value) in value {
+            // For now we are only processing the strength metadata from pending
+            // transaction to live
+            if metadata_key.name() != "strength" {
+                continue;
+            }
+
+            // strength in pending transaction
+            let pending_strength: String =
+                deserialize_scalar::<_, ScalarError>(&metadata_value.clone())
+                    .with_context(|_| error::DeSerializeSnafu {})?;
+
+            let pending_strength: Strength =
+                pending_strength
+                    .parse::<Strength>()
+                    .context(error::ParseSnafu {
+                        strength: pending_strength,
+                    })?;
+
+            // Get the setting strength in live
+            // get_metadata function returns Ok(None) in case strength does not exist
+            // We will consider this case as strength equals strong.
+            let committed_strength: Strength = datastore
+                .get_metadata(metadata_key, key, &Committed::Live)
+                .context(error::DataStoreSnafu { op: "get_metadata" })?
+                .map(|x| x.parse::<Strength>())
+                .transpose()
+                .context(error::TransposeSnafu)?
+                .unwrap_or_default();
+
+            // The get key funtion returns Ok(None) in case if the path does not exist
+            // and error if some path exist and some error occurred in fetching
+            // Hence we we will return error in case of error
+            // from get key function and continue to add/change to weak key
+            // if the value is None.
+            let value = datastore
+                .get_key(key, &Committed::Live)
+                .context(error::DataStoreSnafu { op: "get_key" })?;
+
+            trace!(
+                "check_constraints: key: {:?}, metadata_key: {:?}, metadata_value: {:?}",
+                key.name(),
+                metadata_key.name(),
+                metadata_value
+            );
+
+            match (pending_strength, committed_strength) {
+                (Strength::Weak, Strength::Strong) => {
+                    // Do not change from strong to weak if setting exists
+                    // otherwise commit strength metadata with value as "weak"
+                    if value.is_some() {
+                        return Ok(ConstraintCheckResult::Reject(
+                            "Cannot change setting strength from strong to weak".to_string(),
```

Comment:
We should probably include the key name that is at fault in this error message.

```suggestion
                            format!("Cannot change setting {} strength from strong to weak", key.name()),
```